### PR TITLE
task13

### PR DESCRIPTION
**Pull request by yana-may25**

### CODE REVIEW

**Comment by MrOlegKravchenko** [comment]
Awesome.

**Comment by MrOlegKravchenko** [comment]
I'd like you to declare _length_ method other way.
And define _plus_ and _minus_ methods out of Vec.

File: src/task13/task13.js [hunk 1]
```diff
@@ -1,22 +1,25 @@
...
-    get length() {
-        return (Math.sqrt(Math.pow(this.x, 2) + Math.pow(this.y, 2)));
-    }
-    plus(vector) {
-        return new Vec(this.x + vector.x, this.y + vector.y);
-    }
-    minus(vector) {
-        return new Vec(this.x - vector.x, this.y - vector.y);
-    }
+function Vec(x, y) {
```
Comment (MrOlegKravchenko) [inline src/task13/task13.js:1]: What functions start with capital letter?

File: src/task13/task13.js [hunk 2]
```diff
@@ -1,22 +1,25 @@
...
+    this.x = x;
+    this.y = y;
+}
+
+Vec.prototype.length = function () {
```
Comment (MrOlegKravchenko) [inline src/task13/task13.js:6]: I'd like you to _define_ length method other way in scope of **vec** function :)
And to make it easy to read could you please call it vector?

Comment (MrOlegKravchenko) [reply src/task13/task13.js:6]: or victor

File: src/task13/task13.js (hunk 1 above)
Comment (yana-may25) [reply src/task13/task13.js:1]: what d'you mean
constructors do, don't they?

Comment (yana-may25) [reply src/task13/task13.js:1]: like
function Rabbit(type) {
  this.type = type;
}

Comment (MrOlegKravchenko) [reply src/task13/task13.js]: Exactly

File: src/task13/task13.js (hunk 2 above)
Comment (MrOlegKravchenko) [reply src/task13/task13.js]: I'd like you to **define** _length_ method other way in scope of Vector function

**Comment by MrOlegKravchenko** [comment]
Remove _class_
Try _Object.defineProperty_

**Comment by yana-may25** [comment]
ok, so I've used defineProperty, but it also works if I define a _class_ Vector, not a function. What's wrong with the class?

File: src/task13/task13.js [hunk 3]
```diff
@@ -1,14 +1,12 @@
-class Vector {
-    constructor(x, y) {
-        this.x = x;
-        this.y = y;
-    }
-
-    length() {
-        return (Math.sqrt(Math.pow(this.x, 2) + Math.pow(this.y, 2)));
-    };
+function Vector(x, y) {
```
Comment (MrOlegKravchenko) [inline src/task13/task13.js]: Interesting moment here was using **this**
```
function vector(x, y) {
    this.x = x;
    this.y = y;
    Object.defineProperty(**this**, 'length', {
        get: () => Math.sqrt( this.x * this.x + this.y * this.y )
    });
}
```

**Comment by MrOlegKravchenko** [comment]
Correct and merge

**Pull request merged**